fix(funnels): all events with events

mariusandra opened the pull request:
## Problem

https://posthoghelp.zendesk.com/agent/tickets/3610

A situation like this:

![2023-06-20 23 01 35](https://github.com/PostHog/posthog/assets/53387/be20f0a7-936a-4b29-8f18-d5d009252ae4)

where there is one event with a name in the list, would add `and event in ['$pageview']` to the end of the query

## Changes

Now it adds `1 == 1`

## How did you test this code?

Added a test. Verified the bug works in the UI

mariusandra committed:
fix(funnels): all events with events

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

mariusandra committed:
Merge branch 'master' into fix-funnel=all=events

thmsobrmlr reviewed (changes requested):
This breaks the fix here https://github.com/PostHog/posthog/pull/16025, for which I should definitely have added a regression test.

```json
{
  "kind": "InsightVizNode",
  "source": {
    "kind": "FunnelsQuery",
    "dateRange": {
      "date_from": "-7d"
    },
    "series": [
      {
        "kind": "EventsNode",
        "event": "$pageview",
        "name": "$pageview",
        "math": "total"
      },
      {
        "kind": "EventsNode",
        "event": null,
        "math": "total"
      }
    ],
    "funnelsFilter": {
      "funnel_viz_type": "steps",
      "funnel_order_type": "unordered"
    }
  },
  "full": true
}
```

mariusandra committed:
fix any event regression

mariusandra committed:
fix frontend error display

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

mariusandra committed:
add regression test

mariusandra commented:
I fixed that regression, and added a test now.

I also fixed another annoyance. When erroring, I'd see this:

<img width="1318" alt="image" src="https://github.com/PostHog/posthog/assets/53387/489c4d65-e64a-4e19-91e8-205a98702c16">

... though a true error would be much more informative. So now it'll show:

<img width="1326" alt="image" src="https://github.com/PostHog/posthog/assets/53387/989203bc-2ef5-4948-8400-282411a7d81d">

mariusandra committed:
Merge branch 'fix-funnel=all=events' of github.com:PostHog/posthog into fix-funnel=all=events

mariusandra committed:
Merge branch 'master' into fix-funnel=all=events

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

thmsobrmlr reviewed (approved):
Thanks for adding the regression test and the fix for displaying funnel errors!
